### DISCUSSION TITLE
Support examples in private repositories

### DISCUSSION THREAD

**Discussion by crash7** [vercel/next.js#32787] [Ideas]
# Feature request

Hello, I was about to fork the `create-next-app` package in order to be able to use it with some private repositories but maybe others have this use case too and I can contribute directly to the repository if this gets traction.

## Is your feature request related to a problem? Please describe.

The main use case is to be able to create-next-app's using examples stored in private repositories that a user can have access through github.

## Describe the solution you'd like

My plan is to add another flag to the CLI that allows the user to pass a github personal token.
Then the token will be used in the `got` requests (ie: [here](https://github.com/zeit/next.js/blob/canary/packages/create-next-app/helpers/examples.ts#L16), [here](https://github.com/zeit/next.js/blob/canary/packages/create-next-app/helpers/examples.ts#L30) among other places)

## Describe alternatives you've considered

Personal tokens need to be crafted by going into settings, it would be nice to use the OAuth login but dealing with that could be more problematic/tedious to implement.

## Additional context

[Alternative Github Auth methods docs](https://developer.github.com/v3/auth/)

**Comment by chuabingquan** [comment]
Hello, I can give this a shot! So all the user needs to do is supply a token under a flag, and their list of repositories would show up? **Edit - @crash7 has an implementation ready**

**Comment by crash7** [comment]
Hi! I have this in a fork already, I was waiting for some feedback from the nextjs team in order to send the PR (I didn't see the new labels, they don't trigger a notification).

I added a `githubToken` flag that expects a token in the form of `{githubUser}:{personalToken}`.

Wonder if that's ok

**Reply by brentos99** [reply]
@crash7 did you feature go anywhere? I'm keen to use it...

**Reply by cezarneaga** [reply]
can you give me a link to your fork? I'd like to make a push on this issue

**Comment by matheusrocha89** [comment]
Is anything new related to this discussion?
